pkg/channel/http/http_channel.go

```diff
@@ -167,8 +167,13 @@ func (h *Channel) invokeMethodV1(ctx context.Context, req *invokev1.InvokeMethod
 	channelReq := h.constructRequest(ctx, req)
 
 	if h.ch != nil {
-		h.ch <- 1
+		h.ch <- struct{}{}
 	}
+	defer func() {
+		if h.ch != nil {
+			<-h.ch
+		}
+	}()
```

Comment:
Seems like we should get a test for this. Is that possible?

Reply:
Added test